pkg/chartsync/download.go

```diff
@@ -49,6 +35,20 @@ func ensureChartFetched(base string, source *helmfluxv1.RepoChartSource) (string
 	return chartPath, nil
 }
 
+// makeChartPath gives the expected filesystem location for a chart,
+// without testing whether the file exists or not.
+func makeChartPath(base string, source *helmfluxv1.RepoChartSource) string {
+	// We don't need to obscure the location of the charts in the
+	// filesystem; but we do need a stable, filesystem-friendly path
+	// to them that is based on the URL.
+	repoPath := filepath.Join(base, base64.URLEncoding.EncodeToString([]byte(source.CleanRepoURL())))
+	if err := os.MkdirAll(repoPath, 00750); err != nil {
+		panic(err)
```

Comment:
I know you just moved this but I am generally against panics. It's OK to exit, but with a friendlier error than a stack trace